csm_web/scheduler/admin.py

```diff
@@ -49,273 +88,311 @@ def has_delete_permission(self, request, obj=None):
+            courses = get_visible_courses(request.user)
+            if db_field.name == "section":
+                kwargs["queryset"] = Section.objects.filter(course__in=courses)
+        return super().formfield_for_foreignkey(db_field, request, **kwargs)
+
+    def get_queryset(self, request):
+        qs = super().get_queryset(request)
+        if request.user.is_superuser:
+            return qs
+        return qs.filter(section__course__in=get_visible_courses(request.user))
+
+    def drop_students(self, request, queryset):
+        queryset.update(active=False)
+
+    drop_students.short_description = (
+        "Drop student(s) from a section. This action is reversible, but it is still strongly "
+        "recommended that you double check the name/email/section/course first."
+    )
+
+    def undrop_students(self, request, queryset):
+        queryset.update(active=True)
+
+    undrop_students.short_description = "Re-enroll student(s) in the section they were previously dropped from."
+
+    # Custom fields
+
+    def get_email(self, obj):
+        return obj.user.email
+
+    get_email.short_description = "Email"
+
+    def get_user(self, obj):
+        return get_admin_link_for(obj.user, "admin:scheduler_user_change")
+
+    get_user.short_description = "User"
+
+    # TODO handle nullable section
+    def get_course(self, obj):
+        return get_admin_link_for(obj.section.course, "admin:scheduler_course_change")
+
+    get_course.short_description = "Course"
+
+    def get_mentor(self, obj):
+        return get_admin_link_for(obj.section.mentor, "admin:scheduler_mentor_change")
+
+    def get_attendances(self, obj):
+        attendance_links = []
+        for attendance in obj.attendance_set.all():
+            admin_url = reverse(
+                "admin:scheduler_attendance_change", args=(attendance.id,)
+            )
+            attendance_links.append(
+                format_html(
+                    '<a style="display: block" href="{}">Date {}: {}</a>',
+                    admin_url,
+                    attendance.date,
+                    attendance.presence if attendance.presence else "--",
+                )
+            )
+        attendance_links.sort()
+        return format_html("".join(attendance_links))
+
+    get_attendances.short_description = "Attendances"
+
+
+@admin.register(Mentor)
+class MentorAdmin(CoordAdmin):
+    # list_filter = ("get_course", "active")
+    list_display = ("name", "get_email", "get_course", "get_section")
+    search_fields = ("user__email", "user__first_name", "user__last_name")
+    actions = ("deactivate_profiles", "activate_profiles")
+    # autocomplete_fields = ("section", "user")
+
+    def has_delete_permission(self, request, obj=None):
+        return (
+            False
+        )  # delete sections + mentor by deleting sections, drop by deactivating
+
+    def get_queryset(self, request):
+        qs = super().get_queryset(request)
```

Comment:
Change `qs` to `queryset` for clarity and consistency, preload reload relations (explained above for student case)
```suggestion
        queryset = super().get_queryset(request).select_related('user', 'section__course')
```